### ISSUE TITLE
Assess accessibility for TimelineJS3 and create sub-tickets to implement

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/TimelineJS3#270]
We should identify key features to make TimelineJS3 as accessible as possible. Here are some things we know, more or less:

keyboard navigation: probably with an option to disable, for those who find it problematic. At least left/right arrow to navigate. What about tabbing through zoom in/zoom out/back to beginning? Tab to select a marker? 

@zachwise has done some work on a high-contrast theme. If we add this, we also need to implement theme selectability in general, which we should anyway.

alt-text -- this is hard to manage with spreadsheets, because there are so many fields. If we make progress on the Google Add-on (see #198) maybe we could open up more columns?

**Comment by JoeGermuska** [comment]
keyboard navigation has come up frequently in recent support tickets and should probably be moved ahead on the priority list.

**Comment by jdeezie** [comment]
Hi Joe. I'm wondering if improving the accessibility of the product is still something that is on your radar?

**Comment by terrill** [comment]
There are several fairly simple improvements that could be made to address accessibility. Should I add each of these as individual new tickets?  

**1. Previous/Next buttons aren't accessible.** 
This is closed issue #551 - I could add these details to that issue but it should be reopened. 
Specifically:  
- The buttons (div.tl-slidenav-next and div.tl-slidenav-previous) should be `<button>` elements (or have role="button" and tabindex="0"). Otherwise, they aren't accessible at all to keyboard users.   
 - They should have readable labels for screen reader users and speech input users 
   (e.g., aria-label="Next slide" is accessible to screen reader users; title="Next slide" makes the command discoverable by speech input users, so they can operate the controls by saying "Click next slide") 
 
**2. Heading structure is reversed**  
  Currently, each date is tagged as an `<h3>` (h3.tl-headline-date), and is immediately followed by the headline, which is tagged as `<h2>` (h2.tl-headline). These are opposite what they should be. 

**3. Timeline markers are only accessible to mousers** 
Hover gives them clear visible focus and click advances to the next slide, but neither of these functions is accessible without a mouse. To fix, add role="button" and tabindex="0" to each div.tl-timemarker-content-container

**4. Contrast is too low on timeline markers** 
In order to meet [WCAG 2.0](https://www.w3.org/TR/WCAG20/) contrast requirements at Level AA (the baseline requirement for many accessibility laws and policies), the foreground:background ratio must be 4.5:1. There are many free tools available for testing color combinations to see if they meet this requirement. 

**5. Add clear visible focus indicator** 
Currently this need is irrelevant since nothing is keyboard-accessible. However, once that problem is fixed, a prominent visible focus indicator should be added on :focus. (e.g., a solid blue border).

**Comment by Tanvez** [comment]
When arrowing through the timeline bar, the slide does not update, but when the slide arrows are used to navigate between the slides then both the timeline bar and the slide advance.
Example video:
https://github.com/NUKnightLab/TimelineJS3/assets/29421667/61167280-8c8f-4e17-af4d-1da63b7bd9a8

**Comment by JoeGermuska** [comment]
This is as expected. See #766 

Arrowing through the slides is an explicit "next" or "previous" gesture, causing the display to move to the appropriate adjacent slide.

Arrowing through the timeline bar (we call it the TimeNav) is navigating a list of options, and the return key indicates a choice from the options, equivalent to clicking directly on the selected element. 

This ticket should have been closed when the [Accessibility 2022 milestone](https://github.com/NUKnightLab/TimelineJS3/milestone/7?closed=1) was reached.
